### ISSUE TITLE
[Phase 2][ML] Build LSTM neural network for missing skills prediction

### ISSUE THREAD

**Issue by Ayusohm432** [Ayusohm432/AI-Skills-Gap-Analyzer#18]
## Description
Create `ml_training/train_missing_skills_lstm.py` for multi-label missing skill prediction.

## Architecture
- Input: (batch, 20, 384) — 20 padded skills × BERT embeddings
- LSTM Layer 1: 128 units, dropout=0.2
- LSTM Layer 2: 64 units, dropout=0.2
- Dense: 256 ReLU → 50 sigmoid output
- Loss: Binary Crossentropy, Optimizer: Adam (lr=0.001)
- 50 epochs with early stopping (patience=5)

## Success Metrics
- Recall@10: >0.75
- Recall@20: >0.85
- MRR: >0.8
- Inference latency: <100ms

## Acceptance Criteria
- [x] Model saved to `ml_models/v1.0/missing_skills_lstm.h5`
- [x] Scaler + vocabulary saved alongside model
- [x] Early stopping implemented
